test/FluentMigrator.Tests/Integration/MigrationRunnerTests.cs

```diff
@@ -1980,14 +1969,23 @@ public override void Down()
         {
             Delete.Index("IX_Users_GroupId").OnTable("Users").InSchema("TestSchema").OnColumn("GroupId");
             Delete.Table("Users").InSchema("TestSchema");
-            Delete.Schema("TestSchema");
+            IfDatabase(t => t != "SQLite").Delete.Schema("TestSchema");
+
+            // Can't actually detatch SQLite DB here as migrations run in a transaction
```

Comment:
~I don't think users should be attaching or detaching databases (schemas) in a migration.  I would just remove this comment and the commented out code unless I'm missing something.~ Nevermind.  I understand.  The Up migration is attaching as a side-effect.  It would be ideal of the attach happened at the connection-level.  But this test does raise an interesting scenario of being able to use IfDatabase to specialize behavior.

If pool is disabled, I assume that the database will auto-detach if running in TransactionPerMigration mode?

Reply:
I'm not entirely sure as the SQLite docs to clearly state the behaviour, but given attached databases are attached per connection, I would assume yes, that on closing the connection the attached database is auto-detached.